src/sensor/ping360.cpp

```diff
@@ -37,15 +37,18 @@ Ping360::Ping360()
 
     // Add timer for worst case scenario
     _timeoutProfileMessage.setInterval(_sensorTimeout);
+    _preConfigurationTimer.setInterval(100);
 
     connect(&_timeoutProfileMessage, &QTimer::timeout, this, [this] {
         qCWarning(PING_PROTOCOL_PING360) << "Profile message timeout, new request will be done.";
-        // Since ping360 uses automatic baudrate detection
-        // it's necessary to start the connection to force baud rate changes
-        link()->startConnection();
         requestNextProfile();
     });
 
+    connect(&_preConfigurationTimer, &QTimer::timeout, this, [this] {
+        qCWarning(PING_PROTOCOL_PING360) << "Device Info timeout";
+        startPreConfigurationProcess();
```

Comment:
startPreConfigurationProcess is called multiple times, I think the name is not corresponding to what it does. It is currently running one step of the detection process, right?

Reply:
Indeed, I'll split it.